base/server/python/pki/server/cli/migrate.py

```diff
@@ -601,3 +602,48 @@ def create_link(self, instance, source, dest):
 
         os.symlink(source, dest)
         os.lchown(dest, instance.uid, instance.gid)
+
+    def migrate_service(self, instance):
+        self.migrate_service_java_home(instance)
+
+    def migrate_service_java_home(self, instance):
+        # When JAVA_HOME in the Tomcat service config differs from the
+        # value in /usr/share/pki/etc/pki.conf, update the value in
+        # the service config.
+
+        if "JAVA_HOME" not in os.environ or not os.environ["JAVA_HOME"]:
+            logger.debug("Refusing to migrate JAVA_HOME with missing environment variable")
+            return
+
+        java_home = os.environ["JAVA_HOME"]
+
+        # Update in /etc/sysconfig/<instance>
+        result = self.update_java_home_in_config(instance.service_conf, java_home)
+        self.write_config(instance.service_conf, result)
+
+        # Update in /etc/pki/<instance>/tomcat.conf
+        result = self.update_java_home_in_config(instance.tomcat_conf, java_home)
+        self.write_config(instance.tomcat_conf, result)
+
+    def update_java_home_in_config(self, path, new_value):
+        result = []
+
+        target = "JAVA_HOME="
+        comment = "JAVA_HOME shouldbe set in /etc/pki/pki.conf instead."
+
+        with open(path, 'r') as conf_fp:
+            for line in conf_fp:
+                if not line.startswith(target):
+                    result.append(line)
+                else:
+                    comment_line = '# ' + comment + '\n'
+                    result.append(comment_line)
+                    new_line = '#' + target + '"' + new_value + '"\n'
```

Comment:
I'd add a space after `#` for readability.

Reply:
I think the new value is already stored in `/etc/pki/pki.conf` so we don't have to store it here again, which could become incorrect if we change Java path again in the future. So it might be better to just comment out the old value.